### ISSUE TITLE
Add icons to device automations picker

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/home-assistant-polymer#4274]
Below is a screenshot of what the device automations picker looks like currently. It displays the device name and area which is great. 

![device-automations](https://i.imgur.com/LZI0jTY.png)

But it's lacking icons, which makes it a little harder to quickly pick out a device since you can't really scan the page quickly. For example if you have a dozen devices in your living room and they're also assigned to the living room area, then you'd be seeing the same "Living room" text two dozen times. 

If icons were shown here, you'd be able to quickly skim the list for the chromecast logo or the playstation logo etc and find/select your device a lot faster without having to read much.

Most other sections of Home Assistant display icons for this style of combo/multi-line dropdown. The states and services panels, the entity picker on Lovelace card editors, the device picker on Person editor, etc. I think that it would be useful here too.

**Comment by bramkragten** [comment]
So you suggest to add the functionality to add icons to a device?
Or do you suggest we pick an icon from one of the entities of that devices?

A device doesn't have an icon because it is not used anywhere in the UI besides the picker/config, as all the others are states, they can be used in the UI.

**Comment by SeanPM5** [comment]
I probably explained it poorly, I meant to simply display icons in the dropdown like some other parts of HA already do. 

For example here is the Services panel with a very similar multi-line style and it displays icons to the left of each entity. I think that this would be nice when selecting a device in the Automations editor.

![service](https://i.imgur.com/KLhTxnS.jpg)

I know this screenshot is of switches and not devices, it's just for demonstrating the benefits of showing icons. You could cover all the text on the right side of this image with your hand and still be able to identify which device is which based solely on the icon. 

Showing icons helps to find your device more easily and quickly among a list of potentially dozens of devices.

**Comment by bramkragten** [comment]
Yeah, but I mean is there is no icon for a device now. Just for entities.

**Comment by SeanPM5** [comment]
Ah, I see what you mean now.

In many cases the devices only have a single entity inside them - individual Hue lights and Google Home speakers for example show as unique devices. So in those cases it should be fine to just pick the entity icon to use as a device icon.

But I'm not so sure about how to handle devices with numerous entities, that is a bit more tricky.. I think picking an icon at random would not always be ideal cause it would probably pick weird ones sometimes. Probably would be better to add functionality to specify a device icon. Which now that I remember balloob kind of floated that idea already a few weeks back here: https://github.com/home-assistant/home-assistant-polymer/issues/3021#issuecomment-528617466

**Comment by matthiasdebaat** [comment]
Great solution to use the entity icon when there's only one entity. Devices with numerous entities can use the most general icon `mdi:devices`. When we do this we should make it possible to config a icon on the device edit page.
